code/fundamentals.rb

```diff
@@ -8,8 +8,26 @@
 
 # 1. Create a variable `my_favorite_animal` and assign it the name of your favorite animal
 
+my_favorite_animal = "panda"
+
 # 2. Print "My favorite animal is <your-favorite-animal>" by printing a string with `my_favorite_animal` interpolated into it.
 
+puts "My favorite animal is a #{my_favorite_animal}."
+
 # 3. Write a method `make_noise` which prints the string "BANG!"
 
+def make_noise
+  puts "Bang!"
+end
+
 # 4. Write a method `count_down` which takes one argument and prints every number from it to 0. Assume `count_down` will only be called with positive integers
+ def count_down(i)
+    loop do
+      i -= 1
+      puts i
+      if i == 0
```

Comment:
`def`, `loop` and `if` need to be terminated with `end` for a total of 3 with this method. The `if` is missing it's corresponding `end` keyword.

Your first `end` lines up visually with `loop`, but when ruby reads the code, it matches the first `end` it sees to the most recent code waiting to complete, `if`. It goes on to match the next `end` with `loop` and never sees an `end` to match up with `def`. That's why the method didn't work when you tried invoking it. It was never finished.